src/components/listEditForm/listEditForm.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  useState,
+  useRef,
+  useEffect,
+  type FormEventHandler,
+  type CSSProperties,
+  type ChangeEvent,
+  type RefObject,
+} from 'react'
+import classNames from 'classnames'
+import { useColorScheme } from '../../hooks/contexts'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faSquareCheck } from '@fortawesome/free-regular-svg-icons'
+import styles from './listEditForm.module.css'
+
+const FIXED_BUTTON_WIDTH = 72 // px
+
+interface EditFormProps {
+  formRef: RefObject<any>
+  maxTotalWidth: number
+  className?: string
+  title: string
+  onSubmit: FormEventHandler
+}
+
+const ListEditForm = ({
+  formRef,
+  maxTotalWidth,
+  className,
+  title,
+  onSubmit,
+}: EditFormProps) => {
+  const [maxTextWidth, setMaxTextWidth] = useState<number>(
```

Comment:
We can probably get rid of this entirely and just use
```ts
const MAX_TEXT_WIDTH = maxTotalWidth - FIXED_BUTTON_WIDTH - 2
```
Then we change the actual input width in the `useEffect` hook any time `maxTotalWidth` changes.